File: 0739-daily-temperatures/0739-daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        vector<int> result(temperatures.size());
        vector<int> stack(temperatures.size());

        int top = -1;
        for(int i = 0; i < temperatures.size(); i++){
            while(top >= 0 && temperatures[i] > temperatures[stack[top]]){
                result[stack[top]] = i - stack[top];
                top--;
            }
            stack[++top] = i;
        }
        return result;
    }
};

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        result = [0] * (len(temperatures))
        stack = [0] * (len(temperatures))

        top = -1
        for i in range(len(temperatures)):
            while top >= 0 and temperatures[i] > temperatures[stack[top]]:
                result[stack[top]] = i - stack[top]
                top -= 1
            top += 1
            stack[top] = i
        return result